org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/MavenPropertiesIdentifier.java

```diff
@@ -0,0 +1,72 @@
+package org.eclipse.jdt.ls.core.internal;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Enumeration;
+import java.util.Properties;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.m2e.core.embedder.ArtifactKey;
+
+
+public class MavenPropertiesIdentifier {
```

Comment:
Looks like `MavenPropertiesIdentifier` and `MavenCentralIdentifier` share some common interface, how about extracting to an interface such as IMavenArtifactIdentifier?

```java
public interface IMavenArtifactIdentifier {
    public ArtifactKey identify(IPath path, IProgressMonitor monitor);

    public ArtifactKey identify(File file, IProgressMonitor monitor);
}

```

Reply:
Done. We can use `public ArtifactKey identify(IPath path, IProgressMonitor monitor);` in both cases. I've switched some of the methods in the implementation to private to make it more obvious what is API and what are internal helpers.